chia/rpc/wallet_rpc_api.py

```diff
@@ -3140,11 +3131,6 @@ async def dl_track_new(self, request) -> Dict:
         """Initialize the DataLayer Wallet (only one can exist)"""
         if self.service.wallet_state_manager is None:
             raise ValueError("The wallet service is not currently initialized")
-
-        peer: Optional[WSChiaConnection] = self.service.get_full_node_peer()
-        if peer is None:
-            raise ValueError("No peer connected")
```

Comment:
this code snippet appear to have changed behavior. It used to end here if we weren't connected, but in your patch we just keep going. are you sure that's OK?

With your change we'll create the wallet before throwing, whereas before we would skip creating the wallet in the first place when we're not connected

Reply:
Hm true that it wouldn't create the wallet in case there is no peer but since there anyway can only be one wallet and a user/script would re-try this anyway if it fails it probably doesn't matter at what point the wallet gets created? Also its only really relevant for the first usage. I can change it if you think its bad but i feel like its ok.